app/models/container.rb

```diff
@@ -39,6 +39,18 @@ def event_where_clause(assoc = :ems_events)
     end
   end
 
+  def ems_event_filter
+    {
+      "container_name"      => name,
+      "container_namespace" => container_project.try(:name),
+      "ems_id"              => ext_management_system.try(:id),
+    }
```

Comment:
Does this need to be `.compact` or perhaps if any are missing then we return `nil`?  Asking because the `try` could return `nil`, which would then be saying "where ems_id is nil", which is not correct.

Reply:
Good question... I took what it's doing in the `event_where_clause`.  The same problem can happen with the namespace.

Reply:
Yeah - my thought is that if these things are missing, perhaps we shouldn't be able to access them at all?  Or maybe ems_id=>nil is ok (it would be events on archived containers), but not namespace?  Maybe @agrare has some thoughts here.

Reply:
Container type resources never have a `nil` ems_id anymore.  The records are kept around as long as the ems is, and are "archived" by way of having a non-nil `deleted_on` property.  The only resources that can have a `nil` ems_id are `VmOrTemplate` and `Host` models.

There was a period of time when "disconnected" container type resources would have a `nil` ems_id but an `old_ems_id` but we dropped that in favor of keeping the normal ems_id in `db/migrate/20210728143006_remove_old_ems_id_from_container_tables.rb`

If a container* has a nil ems_id it isn't valid to query on just the name+project

Reply:
Actually I don't see a dependent => destroy on containers/container_groups I thought we did that now, let me investigate.

Reply:
Sounds good, I'll drop the `try`... `do or do not... there is no try`

Reply:
@agrare can I rely on `container_project` existing for each of these classes?

If not, do we drop the namespace from the filter? Or use a default "NOT FOUND" or something?  The first will allow you to see events matching the other conditions.  The second will likely result in no records in the timeline.

Reply:
@agrare this is lower priority but when you get a chance, see above ☝️

Reply:
Okay this is what a "deleted" pod on an active EMS looks like:
```
 #<ManageIQ::Providers::Openshift::ContainerManager::ContainerGroup:0x000055ba73f32720
  id: 87,
  ems_ref: "8f5e9b88-62eb-4ae0-9aa4-ddbd0b36c90b",
  name: "collect-profiles-27971685-8pqdp",
  ems_created_on: Wed, 08 Mar 2023 18:45:00.000000000 UTC +00:00,
  resource_version: "4712183",
  restart_policy: "Never",
  dns_policy: "ClusterFirst",
  ems_id: 3,
  container_node_id: 1,
  last_perf_capture_on: nil,
  container_replicator_id: nil,
  ipaddress: "10.217.1.196",
  type: "ManageIQ::Providers::Openshift::ContainerManager::ContainerGroup",
  container_project_id: 65,
  phase: "Succeeded",
  message: nil,
  reason: nil,
  container_build_pod_id: nil,
  created_on: Wed, 08 Mar 2023 18:56:01.758830000 UTC +00:00,
  deleted_on: Wed, 08 Mar 2023 20:20:04.767354000 UTC +00:00,
  old_container_project_id: 65>]
```

NOTE the ems_id is still present.

If I delete my container_manager then all of my ContainerGroup records are deleted
```
>> ContainerGroup.count
=> 0
```

So I stand behind my previous statement 
> Container type resources never have a nil ems_id anymore. 

And should never have questioned it :laughing: I'm still checking I'm sure we have a cascade destroy somewhere in a child class.

If a `ContainerProject` is deleted, then the pods will be "archived" so they will have an active ems_id but they will have a `nil` `container_project_id`.  This is what the `old_container_project_id` is for.  You will always have at least a non-nil `ems_id` + `old_container_project_id`

Reply:
Scratch that last, we archive `ContainerProjects` also so a deleted project will still be around so yes a deleted pod will have a `container_project_id` unless someone manually deletes "archived" inventory.

Reply:
the good thing is we don't actually use this in the UI right now so whatever we do here won't be a regression 🤣 ... I'll drop all the trys and make yoda proud.